packages/components/upload/src/upload.ts

```diff
@@ -123,6 +128,9 @@ export const uploadBaseProps = buildProps({
     type: definePropType<UploadUserFile[]>(Array),
     default: () => mutable([] as const),
   },
+  'onUpdate:fileList': {
+    type: definePropType<UploadUserFile[]>(Function),
+  },
```

Comment:
Why declare it in both props and emits? 🤔

Reply:
Since `file-list` supports `one-way` and `two-way` modes, `vue` will automatically pass in this `prop` in `two-way` mode, and use this to determine whether it is currently `two-way` or `one-way`.
Learn from `visible` in `tooltip`.

Reply:
I canceled this part.